app/storage/data_access.py

```diff
@@ -0,0 +1,188 @@
+        if returned_data:
+            model, _ = schema.load(returned_data)
+            setattr(model, '_use_dynamo', True)
+        else:
+            logger.debug(
+                'could not find item in dynamodb',
+                table_name=table_name,
+                key_value=key_value)
+
+    if not model:
+        # find in RDS
+        sql_model = config['sql_model']
+        if not sql_model:
+            return
+
+        returned_data = sql_model.query.filter_by(**key).first()
+        if returned_data:
+            model = returned_data.to_new_model()
+            setattr(model, '_use_dynamo', False)
+
+    return model
+
+
+def put(model, overwrite=True):
+    """Inserts or updates the given app model
+
+    :param model: the app model to be saved
+    :param overwrite: if true then the object may overwrite an existing object
+        with the same ID. if not a `ItemAlreadyExistsError` will be raised
+    """
+    config = TABLE_CONFIG[type(model)]
+    schema = config['schema'](strict=True)
+
+    if getattr(model, '_use_dynamo', _is_dynamodb_write_enabled(config)):
+        item, _ = schema.dump(model)
+
+        table_name = _get_table_name(config)
+        try:
+            dynamo_api.put_item(table_name, item, overwrite=overwrite)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ConditionalCheckFailedException':
+                raise ItemAlreadyExistsError() from e
+
+            raise
+    else:
+        if config['sql_model']:
+            sql_model = config['sql_model'].from_new_model(model)
+
+            try:
+                # pylint: disable=maybe-no-member
+                if overwrite:
+                    models.db.session.merge(sql_model)
+                else:
+                    models.db.session.add(sql_model)
+
+                models.db.session.commit()
+            except IntegrityError as e:
+                raise ItemAlreadyExistsError() from e
+
+
+def delete(model):
+    """Deletes the given app model
+
+    :param model: the app model to be deleted
+    """
+    config = TABLE_CONFIG[type(model)]
+    key_name = config['key_field']
+    key = {key_name: getattr(model, key_name)}
+
+    if getattr(model, '_use_dynamo', _is_dynamodb_write_enabled(config)):
+        table_name = _get_table_name(config)
+        dynamo_api.delete_item(table_name, key)
+    else:
+        if config['sql_model']:
+            sql_model = config['sql_model'].from_new_model(model)
+
+            # pylint: disable=maybe-no-member
+            sql_model = models.db.session.merge(sql_model)
+            models.db.session.delete(sql_model)
+            models.db.session.commit()
+
+
+def setup_tables():
+    """Sets up DynamoDB tables
+    """
+    for config in TABLE_CONFIG.values():
+        if _is_dynamodb_read_enabled(config):
+            table_name = _get_table_name(config)
+            if table_name:
+                dynamo_api.create_table(table_name, config['key_field'])
```

Comment:
Do we want this? as we create our tables with Terraform.

Reply:
Creates tables in mock dynamodb